src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UsesSignedCommits.java

```diff
@@ -0,0 +1,162 @@
+
+  /**
+   * 1 year in millis.
+   */
+  private static final long DELTA = 365 * 24 * 60 * 60 * 1000L;
+
+  /**
+   * The percentage threshold of number of signed commits against number of total commits.
+   */
+  private static final double SIGNED_COMMIT_PRECENTAGE_THRESHOLD = 0.4;
+
+  /**
+   * A git token provided by user to access the Github API.
+   */
+  private final String gitToken;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param where A GitHub organization of user name.
+   * @param name A name of a repository.
+   * @param github An interface to the GitHub API.
+   * @param gitToken provided by user to access GitHub API.
+   */
+  public UsesSignedCommits(String where, String name, GitHub github, String gitToken)
+      throws IOException {
+    super(where, name, github);
+    this.gitToken = gitToken;
+  }
+
+  @Override
+  public UsesSignedCommits update(ValueSet values) throws IOException {
+    Objects.requireNonNull(values, "Hey! Values can't be null!");
+    System.out.println("[+] Figuring out if the project uses any signed commits ...");
+
+    Optional<Value> something = cache().get(url, USES_VERIFIED_SIGNED_COMMITS);
+    if (something.isPresent()) {
+      values.update(something.get());
+      return this;
+    }
+
+    boolean flag = densityEvaluation();
+    if (callback.canTalk()) {
+      String question = String.format(
+          "Does project %s/%s use any signed commits? Say yes, no, or skip, please.",
+          where, name);
+
+      Answer answer = new YesNoSkipQuestion(callback, question).ask();
+      switch (answer) {
+        case YES:
+          flag = true;
+          break;
+        case NO:
+          flag = false;
+          break;
+        default:
+          throw new IllegalArgumentException(
+              String.format("Hey! You gave me an unexpected answer: %s", answer));
+      }
+    }
+
+    Value<Boolean> signedCommits = new BooleanValue(USES_VERIFIED_SIGNED_COMMITS, flag);
+    values.update(signedCommits);
+    cache().put(url, signedCommits, tomorrow());
+
+    return this;
+  }
+
+  /**
+   * It gathers the list of commits within one year from the current date. It checks if these
+   * commits are signed or not. Then it calculates density ratio between number of signed commits vs
+   * unsigned commits. If the density ratio >= 1, then we set the flag = true else flag = false.
+   * 
+   * @return boolean value indicating if the project uses signed commit.
+   * @throws IOException
+   * 
+   */
+  private boolean densityEvaluation() throws IOException {
+    GHRepository repository = github.getRepository(path);
+    double signedCounter = 0;
+    double counter = 0;
+    Date date = new Date(System.currentTimeMillis() - DELTA);
+
+    for (GHCommit commit : repository.listCommits()) {
+      if (commit.getCommitDate().after(date)) {
+        counter++;
+        // TODO: Remove this commit rest call when org.kohsuke:github-api implements
```

Comment:
I would say that we may consider merging this data provider with `FirstCommit` and `ProjectStarted` data providers since all of them use commit history. That would speed up gathering data. Could you please update the TODO and open an issue for this improvement?

Reply:
#69